packages/protocol/src/utils/SPID.ts

```diff
@@ -0,0 +1,176 @@
+import { format } from 'util'
+
+class SPIDValidationError extends Error {
+    data: SPIDish
+    constructor(msg: string, data: SPIDish) {
+        super(format(msg, data))
+        this.data = data
+        if (Error.captureStackTrace) {
+            Error.captureStackTrace(this, this.constructor)
+        }
+    }
+}
+/**
+ * SPID - Stream Partition ID
+ */
+// if you're using JS numbers for
+// ids or keys going over MAX_SAFE_INTEGER has the potential for disasterous
+// effects e.g. serving private user information
+export default class SPID {
+    /** stream id */
+    public readonly id: string
```

Comment:
If we rename this to `streamId`, we don't need the comment, and maybe not even the `split` method as object destructuring will have clear and unique names?